Userland/Utilities/groups.cpp

```diff
@@ -0,0 +1,64 @@
+#include <grp.h>
+#include <pwd.h>
+#include <unistd.h>
+
+#include <AK/Vector.h>
+#include <LibCore/Account.h>
+#include <LibCore/ArgsParser.h>
+
+static void print_gid_list(const Vector<gid_t>& gids)
+{
+    for (auto gid : gids) {
+        auto* gr = getgrgid(gid);
+        printf("%s ", gr->gr_name);
+    }
+    printf("\n");
+}
+
+int main(int argc, char** argv)
+{
+    if (unveil("/etc/passwd", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (unveil("/etc/group", "r") < 0) {
+        perror("unveil");
+        return 1;
+    }
+
+    if (pledge("stdio rpath", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    Vector<const char*> usernames;
+
+    Core::ArgsParser args_parser;
+    args_parser.set_general_help("Print group memberships for each username or, if no username is specified, for the current process.");
+    args_parser.add_positional_argument(usernames, "Usernames to list group memberships for", "usernames", Core::ArgsParser::Required::No);
+    args_parser.parse(argc, argv);
+
+    if (usernames.is_empty()) {
+        auto uid = geteuid();
+        auto* passwd = getpwuid(uid);
+        if (!passwd) {
+            fprintf(stderr, "Unknown user id '%d'\n", uid);
+            return 1;
+        }
+        auto gids = Core::Account::get_gids(passwd->pw_name);
```

Comment:
I think the API here should have been `Account::from_name(...)->extra_gids()` instead of exposing this internal method?